test/membrane_framerate_converter_test.exs

```diff
@@ -0,0 +1,122 @@
+defmodule Membrane.FramerateConverterTest do
+  @moduledoc """
+  Tests for FramerateConverter module.
+  """
+
+  use ExUnit.Case
+  use Membrane.Pipeline
+
+  import Membrane.Testing.Assertions
+
+  alias Membrane.File.{Sink, Source}
+  alias Membrane.H264.FFmpeg.{Parser, Decoder, Encoder}
+  alias Membrane.Testing
+  alias Membrane.Testing.Pipeline
+
+  require Membrane.Logger
+
+  @fps_test_file Path.expand("fixtures/testsrc_5_fps.h264", __DIR__)
+  @timestamp_test_file Path.expand("fixtures/10-720p.h264", __DIR__)
+  @fps_file_framerate {5, 1}
+  @timestamp_file_framerate {30, 1}
+
+  defp expand_path(file_name) do
+    Path.expand("fixtures/#{file_name}", __DIR__)
+  end
+
+  defp prepare_output(file_name) do
+    output_path = expand_path(file_name)
+
+    File.rm(output_path)
+    on_exit(fn -> File.rm(output_path) end)
+
+    output_path
+  end
+
+  defp assert_files_equal(file_a, file_b) do
+    assert {:ok, a} = File.read(file_a)
+    assert {:ok, b} = File.read(file_b)
+    assert a == b
+  end
+
+  describe "FramerateConverter should" do
+    defp perform_general_test(elements) do
+      pipeline_options = %Pipeline.Options{elements: elements}
+      assert {:ok, pid} = Pipeline.start_link(pipeline_options)
+
+      assert Pipeline.play(pid) == :ok
+      assert_end_of_stream(pid, :sink, :input, 2_000)
+      Pipeline.stop_and_terminate(pid, blocking?: true)
+    end
+
+    defp perform_fps_test(output_filename, reference_filename, target_framerate) do
+      output_path = prepare_output(output_filename)
+      reference_file = expand_path(reference_filename)
+
+      elements = [
+        file: %Source{chunk_size: 40_960, location: @fps_test_file},
+        parser: %Parser{framerate: @fps_file_framerate},
+        decoder: Decoder,
+        converter: %Membrane.FramerateConverter{framerate: target_framerate},
+        sink: %Sink{location: output_path}
+      ]
+
+      perform_general_test(elements)
+      assert_files_equal(output_path, reference_file)
+    end
+
+    test "convert video with given pts" do
+      output_path = prepare_output("out.h264")
+
+      elements = [
+        file: %Source{chunk_size: 40_960, location: @fps_test_file},
+        parser: %Parser{framerate: @fps_file_framerate},
+        decoder: Decoder,
+        converter: %Membrane.FramerateConverter{framerate: {2, 1}},
+        encoder: Encoder,
+        sink: %Sink{location: output_path}
+      ]
+
+      perform_general_test(elements)
+    end
+
+    test "convert video to the lower frame rate correctly" do
+      perform_fps_test("out_2_fps.yuv", "reference-testsrc_2_fps.yuv", {2, 1})
```

Comment:
It would be perfect to also test wild framerates like [NTSC 30_000/1001 fps (~29,97)](https://en.wikipedia.org/wiki/NTSC#Resolution_and_refresh_rate) to make sure all the Ratio calls are OK
Context: Ratio has a property that can bite - 
```elixir
iex(7)> Ratio.new(30_000, 1001)
30000 <|> 1001
iex(8)> Ratio.new(30_000, 1001) |> Map.from_struct
%{denominator: 1001, numerator: 30000}
iex(9)> Ratio.new(30_000, 1000)                   
30
```
So if using 30/1 you will have numbers instead of Ratio structs and the code might fail if receives an actual Ratio struct (it happened before :wink:)